torchvision/datasets/cifar.py

```diff
@@ -62,44 +66,23 @@ def __init__(self, root, train=True,
             raise RuntimeError('Dataset not found or corrupted.' +
                                ' You can use download=True to download it')
 
-        # now load the picked numpy arrays
-        if self.train:
-            self.train_data = []
-            self.train_labels = []
-            for fentry in self.train_list:
-                f = fentry[0]
-                file = os.path.join(self.root, self.base_folder, f)
-                fo = open(file, 'rb')
+        # now load the pickled numpy arrays
+        data_list = self.train_list if train else self.test_list
+        self.data = []
+        self.labels = {k: [] for k in self.label_keys}
+        for fname, _ in data_list:
+            with open(os.path.join(self.root, self.base_folder, fname), 'rb') as fo:
                 if sys.version_info[0] == 2:
                     entry = pickle.load(fo)
                 else:
                     entry = pickle.load(fo, encoding='latin1')
-                self.train_data.append(entry['data'])
-                if 'labels' in entry:
-                    self.train_labels += entry['labels']
-                else:
-                    self.train_labels += entry['fine_labels']
-                fo.close()
-
-            self.train_data = np.concatenate(self.train_data)
-            self.train_data = self.train_data.reshape((50000, 3, 32, 32))
-            self.train_data = self.train_data.transpose((0, 2, 3, 1))  # convert to HWC
-        else:
-            f = self.test_list[0][0]
-            file = os.path.join(self.root, self.base_folder, f)
-            fo = open(file, 'rb')
-            if sys.version_info[0] == 2:
-                entry = pickle.load(fo)
-            else:
-                entry = pickle.load(fo, encoding='latin1')
-            self.test_data = entry['data']
-            if 'labels' in entry:
-                self.test_labels = entry['labels']
-            else:
-                self.test_labels = entry['fine_labels']
```

Comment:
@lucasb-eyer This case is no longer handled for cifar10 then? as now it only checks for `entry['labels']` If I understand the changes correctly.

Reply:
I'm not sure I understand the question. CIFAR10 only has `labels`, there is no such thing as `fine_labels` and `coarse_labels` in CIFAR10.

The removed code you are commenting on only used `labels` in CIFAR10 (which still is used after the changes) and only used `fine_labels` in CIFAR100 (which now is either `fine_labels`, `coarse_labels`, or both).

Not sure if this answers your question.